app/src/main/java/com/keylesspalace/tusky/util/LinkHelper.kt

```diff
@@ -0,0 +1,260 @@
+     * See also : https://github.com/tuskyapp/Tusky/issues/846
+     *            https://github.com/tuskyapp/Tusky/pull/916 */
+    if (end >= builder.length || builder[end] == '\n') {
+        builder.insert(end, "\u200B")
+    }
+}
+
+/**
+ * Returns the first account id with matching username from mentions that isn't contained in usedIds,
+ * or the id of the last matching account, if all matching ids are already contained
+ * @param username the username to match
+ * @param mentions the mentions to search
+ * @param usedIds the collection of ids already used
+ */
+private fun firstUnusedMention(username: String, mentions: Array<Status.Mention>, usedIds: Collection<String>): String? {
+    var id: String? = null
+    for (mention in mentions) {
+        if (mention.localUsername.equals(username, true)) {
+            id = mention.id
+            if (!usedIds.contains(id)) {
+                break
+            }
+        }
+    }
+    return id
+}
+
+private fun getTagSpan(tag: String, listener: LinkListener): ClickableSpan {
+    return object : ClickableSpanNoUnderline() {
+        override fun onClick(widget: View) {
+            listener.onViewTag(tag)
+        }
+    }
+}
+
+private fun getAccountSpan(id: String?, listener: LinkListener): ClickableSpan {
+    return object : ClickableSpanNoUnderline() {
+        override fun onClick(widget: View) {
+            listener.onViewAccount(id)
+        }
+    }
+}
+
+private fun getLinkSpan(url: String, listener: LinkListener): ClickableSpan {
+    return object: CustomURLSpan(url) {
+        override fun onClick(widget: View?) {
+            listener.onViewUrl(url)
+        }
+    }
+}
+
+/**
+ * Put mentions in a piece of text and makes them clickable, associating them with callbacks to
+ * notify when they're clicked.
+ *
+ * @param view the returned text will be put in
+ * @param mentions any '@' mentions which are known to be in the content
+ * @param listener to notify about particular spans that are clicked
+ */
+fun setClickableMentions(view: TextView, mentions: Array<Status.Mention>?, listener: LinkListener) {
+    if (mentions.isNullOrEmpty()) {
+        view.text = null
+        return
+    }
+    val builder = SpannableStringBuilder()
+    var start = 0
+    var end = 0
+    var firstMention = true
+
+    for (mention in mentions) {
+        val accountUsername = mention.localUsername
+        val accountId = mention.id
+        val customSpan = getAccountSpan(accountId, listener)
+
+        end += 1 + accountUsername!!.length // length of @ + username
+        val flags = builder.getSpanFlags(customSpan)
+        if (firstMention) {
+            firstMention = false
+        } else {
+            builder.append(" ")
+            start += 1
+            end += 1
+        }
+        builder.append("@")
+        builder.append(accountUsername)
+        builder.setSpan(customSpan, start, end, flags)
+        builder.append("\u200B") // same reasonning than in setClickableText
```

Comment:
Typo in comment. Also maybe make that `"\u200B"` a constant so its easier to see its a zero-width space?